### PR TITLE
Remove pull API and protocol support

### PR DESCRIPTION
**Pull request by Mallets**
#782 provides a path to implement *last-value* kind of functionality for the pull subscriber. 
While #782 is still in a draft state, a proper PR is expected to come in the near future from @evshary . 
In the meanwhile, I've quickly modified the `z_pull` example to have the exact same behaviour (although with a slightly different API).

### CODE REVIEW

**Comment by imstevenpmwork** [comment]
@Mallets Can you link this to the relevant issue?

File: examples/examples/z_pull.rs [hunk 1]
```diff
@@ -13,60 +13,80 @@
...
-#[derive(clap::Parser, Clone, PartialEq, Eq, Hash, Debug)]
+#[derive(clap::Parser, Clone, PartialEq, Debug)]
 struct SubArgs {
     #[arg(short, long, default_value = "demo/example/**")]
     /// The Key Expression to subscribe to.
     key: KeyExpr<'static>,
+    /// The size of the cache.
+    #[arg(long, default_value = "3")]
+    cache: usize,
+    /// The size of the cache.
```
Comment (OlivierHecart) [inline examples/examples/z_pull.rs:81]: Copy-paste error

File: zenoh/src/net/routing/hat/client/pubsub.rs [hunk 2]
```diff
@@ -132,8 +126,7 @@ fn declare_client_subscription(
     sub_info: &SubscriberInfo,
 ) {
     register_client_subscription(tables, face, id, res, sub_info);
-    let mut propa_sub_info = *sub_info;
-    propa_sub_info.mode = Mode::Push;
+    let propa_sub_info = *sub_info;
```
Comment (OlivierHecart) [inline zenoh/src/net/routing/hat/client/pubsub.rs:129]: I guess this variable is not needed any more.

File: zenoh/src/net/routing/hat/linkstate_peer/pubsub.rs [hunk 3]
```diff
@@ -245,8 +239,7 @@ fn declare_client_subscription(
     sub_info: &SubscriberInfo,
 ) {
     register_client_subscription(tables, face, id, res, sub_info);
-    let mut propa_sub_info = *sub_info;
-    propa_sub_info.mode = Mode::Push;
+    let propa_sub_info = *sub_info;
```
Comment (OlivierHecart) [inline zenoh/src/net/routing/hat/linkstate_peer/pubsub.rs:242]: I guess this variable is not needed any more.

File: zenoh/src/net/routing/hat/p2p_peer/pubsub.rs [hunk 4]
```diff
@@ -132,8 +126,7 @@ fn declare_client_subscription(
     sub_info: &SubscriberInfo,
 ) {
     register_client_subscription(tables, face, id, res, sub_info);
-    let mut propa_sub_info = *sub_info;
-    propa_sub_info.mode = Mode::Push;
+    let propa_sub_info = *sub_info;
```
Comment (OlivierHecart) [inline zenoh/src/net/routing/hat/p2p_peer/pubsub.rs:129]: I guess this variable is not needed any more.

File: zenoh/src/net/routing/hat/router/pubsub.rs [hunk 5]
```diff
@@ -298,8 +291,7 @@ fn declare_client_subscription(
     sub_info: &SubscriberInfo,
 ) {
     register_client_subscription(tables, face, id, res, sub_info);
-    let mut propa_sub_info = *sub_info;
-    propa_sub_info.mode = Mode::Push;
+    let propa_sub_info = *sub_info;
```
Comment (OlivierHecart) [inline zenoh/src/net/routing/hat/router/pubsub.rs:294]: I guess this variable is not needed any more.

File: zenoh/src/subscriber.rs [hunk 6]
```diff
@@ -248,28 +164,11 @@ impl Drop for SubscriberInner<'_> {
...
-impl From<PullMode> for Mode {
-    fn from(_: PullMode) -> Self {
-        Mode::Pull
-    }
-}
-
 /// The mode for push subscribers.
 #[non_exhaustive]
 #[derive(Debug, Clone, Copy)]
 pub struct PushMode;
```
Comment (OlivierHecart) [inline zenoh/src/subscriber.rs:170]: This struct and the SubscriberBuilder parametrization should not be needed any more.

Comment (Mallets) [reply zenoh/src/subscriber.rs]: Done

**Pull request merged**